Mathlib/Topology/Algebra/Monoid.lean

```diff
@@ -280,31 +279,35 @@ theorem HasContinuousMul.of_nhds_one {M : Type u} [Monoid M] [TopologicalSpace M
     have key₂ : ((fun x => x₀ * x) ∘ fun x => y₀ * x) = fun x => x₀ * y₀ * x :=
       by
       ext x
-      simp
+      simp [mul_assoc]
     calc
-      map (uncurry (· * ·)) (𝓝 (x₀, y₀)) = map (uncurry (· * ·)) (𝓝 x₀ ×ᶠ 𝓝 y₀) := by
-        rw [nhds_prod_eq]
-      _ = map (fun p : M × M => x₀ * p.1 * (p.2 * y₀)) (𝓝 1 ×ᶠ 𝓝 1) := by
-        rw [uncurry, hleft x₀, hright y₀, prod_map_map_eq, Filter.map_map]
-      _ = map ((fun x => x₀ * x) ∘ fun x => x * y₀) (map (uncurry (· * ·)) (𝓝 1 ×ᶠ 𝓝 1)) := by
-        rw [key, ← Filter.map_map]
+      map (uncurry (· * ·)) (𝓝 (x₀, y₀)) = map (uncurry (· * ·)) (𝓝 x₀ ×ᶠ 𝓝 y₀) :=
+        by rw [nhds_prod_eq]
+      _ = map (fun p : M × M => x₀ * p.1 * (p.2 * y₀)) (𝓝 1 ×ᶠ 𝓝 1) :=
+        by
+          simp_rw [uncurry]
+          rw [hleft x₀, hright y₀]
+          rw [prod_map_map_eq, Filter.map_map]
+          simp only [Function.comp]
+      _ = map ((fun x => x₀ * x) ∘ fun x => x * y₀) (map (uncurry (· * ·)) (𝓝 1 ×ᶠ 𝓝 1)) :=
+        by rw [key, ← Filter.map_map]
```

Comment:
Any idea what went wrong here?

Reply:
I have no idea why the `uncurry` and the `Function.comp` fail with `rw` (it might be the 'weaker `rw`'-issue) and we need to call `Function.comp` explicitly now because it is not reducible anymore.

Reply:
I've added a porting note for the `rw` to `simp_rw` change